ArmPkg/Drivers/ArmGic/ArmGicLib.c

```diff
@@ -139,10 +139,53 @@ ArmGicSendSgiTo (
   IN  INTN  SgiId
   )
 {
-  MmioWrite32 (
-    GicDistributorBase + ARM_GIC_ICDSGIR,
-    (UINT32)(((TargetListFilter & 0x3) << 24) | ((CPUTargetList & 0xFF) << 16) | SgiId)
-    );      // MS_CHANGE
+  // MU_CHANGE Starts: Added check for supported GIC version and expanded support for sending SGI on GICv3 and GICv4.

```

Comment:
The comments say GICv3 and v4, but the code only calls a GICv3 API. Is it the same API for GICv4?

Reply:
Yeah, the implementation for this function is essentially following the doc: https://developer.arm.com/documentation/198123/0302/Sending-and-receiving-Software-Generated-Interrupts (I did not add it since I am afraid it may disappear someday). But the `ICC_SGI1R_EL1` function was following the existing semantics of existing ArmGicLib, which only has distinction over v2 and v3.